### ISSUE TITLE
Retrieve all attributes with getAttributes() method

### ISSUE THREAD

**Issue by sergtimosh** [wix/Detox#3165]
**Is your feature request related to a problem? Please describe.**
_This particular case is Android specific._
In my tests I rely on such attributes like _'is-checked'_ for toggle value, another attributes are used for  button state - enabled/disabled which currently ommited in the returning object of getAttributes() method. The only way to retrieve this values is parsing error object like it's being described [here](https://www.npmjs.com/package/detox-getprops)

**Describe the solution you'd like**
Yield more attribute-value pairs from the element to extend potential checks.

<img width="672" alt="image" src="https://user-images.githubusercontent.com/26712442/148191511-cdb87d24-25a4-4b3d-9fa7-347e4b43bb7d.png">

**Comment by d4vidi** [comment]
@sergtimosh please revisit our documentation on [getAttributes](https://wix.github.io/Detox/docs/api/actions-on-element/#getattributes). I think fields such as `enabled` and `value` might be what you're looking for.

**Comment by sergtimoshUptarget** [comment]
@d4vidi, unfortunately `value` is relevant only for the iOS version of the app. In order to achieve the same on Android the only option is `is-checked` for my case.
Properties from android element:
`{
  identifier: 'header-results-toggle',
  visibility: 'visible',
  visible: true,
  alpha: 1,
  elevation: 0,
  height: 74,
  width: 128,
  focused: false,
  enabled: true,
  text: '',
  length: 0,
  textSize: 39
}`

**Comment by sergtimosh** [comment]
@d4vidi, unfortunately `value` is relevant only for the iOS version of the app. In order to achieve the same on Android the only option is `is-checked` for my case.
Properties from android element:
`{
  identifier: 'header-results-toggle',
  visibility: 'visible',
  visible: true,
  alpha: 1,
  elevation: 0,
  height: 74,
  width: 128,
  focused: false,
  enabled: true,
  text: '',
  length: 0,
  textSize: 39
}`

**Comment by owens-ben** [comment]
Bump, this would be really helpful

**Comment by noomorph** [comment]
@asafkorem could you look what extras could be exposed on iOS side, please?

**Comment by asafkorem** [comment]
> @asafkorem could you look what extras could be exposed on iOS side, please?

Already did @noomorph, in the corresponding issue for iOS attributes 🙂 
https://github.com/wix/Detox/issues/3116#issuecomment-982008651

**Comment by d4vidi** [comment]
Legit. Let's try to leave the discussion on extra attributes to #3116. I'll take a glance at the fix-potential for Android for *this specifically*, in a bit.

**Comment by d4vidi** [comment]
So while we do [consider the `isChecked` property](https://github.com/wix/Detox/blob/b1bd601e6d7407b7f1866e67542f49c31cfa59ea/detox/android/detox/src/full/java/com/wix/detox/espresso/action/GetAttributesAction.kt#L122) of an Android view in `getAttributes()`, we seem to limit ourselves to `CheckBox` views. The proper fix here is likely to extend things to any `Checkable`, which `CheckBox` implements anyways.

BTW, seems like there are a ton of other classes that do this, so it would be a good idea to also skim through in order to make sure all associated views are also relevant:

<img width="365" alt="image" src="https://user-images.githubusercontent.com/9818880/212048531-8b0f3643-2e90-4416-9a23-b455ee9ed6b3.png">


So to sum things up, in order to fix this, we need to:
1. Extend the `value` property to match any `Checkable`. We need to do this in an override-able way, meaning, that if a class holds some other meaningful attribute that is more tightly associated with the concept of `value` (e.g. a checkable text view), it should make usage of `value` for it.
2. Apply more view types (e.g. toggles, such as specified by the OP) in the E2E tests we have for this, at least on Android.
